### PR TITLE
[DOC] standardise tut/introduction

### PR DESCRIPTION
**Pull request by anbratu**
As mentioned in #1287, here is the first part for discussion.

### CODE REVIEW

**Comment by anbratu** [comment]
According to the plan to have complete code snippets in the tutorial, I wrote a new code file for seq.fasta file read. Travis failed on building this file (introduction_input_file), because he's not able to acess the seq.fasta, which is in the same directory.

I have some ideas, how to handle this:

1) Ignore this problem, because the code is right.
2) Try and catch block around this line: https://github.com/anbratu/seqan3/blob/6f0128778e674672a0122fa80ea5a6523eeed006/doc/tutorial/introduction/introduction_file_input.cpp#L10
3) As in original version, create the file in runetime.

Is there any Preference, how to tackle it?

File: doc/tutorial/introduction/index.md [hunk 1]
```diff
@@ -27,12 +27,15 @@ Most good tutorials start with an easy *Hello World!* program. So have a look:
...
 Most of them are compilable as-is, but some are only valid in their context,
 e.g. they depend on other code snippets given before/after the current one or
-other statements implied by the text. You can copy'n'paste freely from these examples,
-this implies no copyright-obligations (however distributing SeqAn or an application
-using it does, see [Copyright](https://docs.seqan.de/seqan/3-master-user/about_copyright.html) and [Citing](https://docs.seqan.de/seqan/3-master-user/about_citing.html).
+other statements implied by the text. 
+# This information is no longer necessary, because all code snippets will be completed (regarding to agreement with eseiler).
+# But I leave it here, until whole tutorial is done.
+
+You can copy'n'paste freely from the code snippets, this implies no copyright-obligations (however distributing SeqAn or an application using it does, see [Copyright](https://docs.seqan.de/seqan/3-master-user/about_copyright.html) and [Citing](https://docs.seqan.de/seqan/3-master-user/about_citing.html)).
```
Comment (joergi-w) [inline doc/tutorial/introduction/index.md]: Please reduce the line length to 120 characters.

File: doc/tutorial/introduction/index.md [hunk 2]
```diff
@@ -130,7 +132,7 @@ argument to your program (e.g. call `./myprogram seq.fasta`).
 \snippet introduction_read_fasta.cpp read
 \endsolution
 
-Note that the same code can also read FastQ files and the `qual` variable will not be empty then. If you like, try it!
+Missing `include <vector>` and `include <string>`? As shown above, they are already part of seqan3! Note that the same code can also read FastQ files and the `qual` variable will not be empty then. If you like, try it!
```
Comment (joergi-w) [inline doc/tutorial/introduction/index.md]: We have the policy that _in tutorials and snippets_ we state all includes that are needed directly from the functions in the code snippet. Although they can be technically omitted, they demonstrate from which header files the functions or data structures are used. Therefore, please remove this sentence and adapt the includes accordingly.

File: doc/tutorial/introduction/index.md [hunk 3]
```diff
@@ -153,6 +154,8 @@ a lot of parameters for the alignment computation, for instance score functions,
 you wish to compute a traceback or not. The configurations have their own namespace seqan3::align_cfg and can
 be combined via the logical OR operator (`|`) for building combinations. Check out the alignment tutorial if you want to learn more.
 
+Note: In general, we give `using namespace seqan3;` a miss, to show which parts are from SeqAn3 and which arn't. But for better readabiliy we use a `using` declaration for the namespace member `_dna5`. You can give it a miss, if you work with the overall namespace or declare it in lines 20 and 21 directly.
```
Comment (joergi-w) [inline doc/tutorial/introduction/index.md]: Please reduce the line length to 120 characters.

Comment (joergi-w) [reply doc/tutorial/introduction/index.md]: _to give a miss_ is colloquial language. Please use _omit_ or something similar (2x).

Comment (joergi-w) [reply doc/tutorial/introduction/index.md]: ~arn't~ => aren't

Comment (joergi-w) [reply doc/tutorial/introduction/index.md]: ~readabiliy~ => readability

Comment (joergi-w) [reply doc/tutorial/introduction/index.md]: Instead of `Note:` you can use doxygen's `\note` command as in the line below.

**Comment by joergi-w** [comment]
> According to the plan to have complete code snippets in the tutorial, I wrote a new code file for seq.fasta file read. Travis failed on building this file (introduction_input_file), because he's not able to acess the seq.fasta, which is in the same directory.
> 
> I have some ideas, how to handle this:
> 
> 1. Ignore this problem, because the code is right.
> 2. Try and catch block around this line: https://github.com/anbratu/seqan3/blob/6f0128778e674672a0122fa80ea5a6523eeed006/doc/tutorial/introduction/introduction_file_input.cpp#L10
> 3. As in original version, create the file in runetime.
> 
> Is there any Preference, how to tackle it?

Please create the file at run time (version 3).

File: doc/tutorial/introduction/index.md (hunk 2 above)
Comment (anbratu) [reply doc/tutorial/introduction/index.md]: According to this, I added #include <filesystem> in two files. One passed the tests (introduction_file_input), but the other not (introduction_align). I knew it's new in c++ 17, but why did one pass and one not? And am I to skip it or leave it in?

File: doc/tutorial/introduction/introduction_hello_world.cpp [hunk 4]
```diff
@@ -1,10 +1,9 @@
...
-using namespace seqan3;
-
 int main()
 {
-    debug_stream << "Hello World!\n";
+    seqan3::debug_stream << "Hello world\n";
+    return 0;
 }
-//! [hello]
+//! [hello]
```
Comment (joergi-w) [inline doc/tutorial/introduction/introduction_hello_world.cpp]: insert newline at end of file

File: doc/tutorial/introduction/introduction_read_fasta.cpp [hunk 5]
```diff
@@ -1,37 +1,44 @@
...
+        seqan3::debug_stream << "[PARSER ERROR] " << ext.what() << '\n';
         return 0;
     }
-    debug_stream << "Reading file " << filename << "\n";
 
-    // Create the vector to store sequences of type seqan3::dna5_vector.
+    seqan3::debug_stream << "Reading file " << filename << "\n";
+
+    // Create the vector to store sequences of type seqan3::dna5_vector, 
+    // which means elemtens of type seqan3::dna5_vector.
```
Comment (joergi-w) [inline doc/tutorial/introduction/introduction_read_fasta.cpp]: Remove this which... part please, it's confusing.

File: doc/tutorial/introduction/introduction_align.cpp [hunk 6]
```diff
@@ -1,55 +1,53 @@
...
+        seqan3::debug_stream << res.score() << '\n';            // => -4
+        seqan3::debug_stream << res.alignment() << '\n';        // =>       0     .    :
+                                                                //            ACGTGATG--
+                                                                //            | |||||
+                                                                //            A-GTGATACT
     }
-//! [alignment]
     return 0;
 }
+//! [alignment]
```
Comment (joergi-w) [inline doc/tutorial/introduction/introduction_align.cpp]: insert newline at end of file

File: doc/tutorial/introduction/seq.fasta [hunk 7]
```diff
@@ -0,0 +1,4 @@
+>seq1
+ACGTGATG
+>seq2
+AGTGATACT
```
Comment (joergi-w) [inline doc/tutorial/introduction/seq.fasta]: This file is created at run time, so please delete it.

File: doc/tutorial/introduction/introduction_file_input.cpp [hunk 8]
```diff
@@ -0,0 +1,55 @@
+//! [fileinput_1]
+#include <string>
+#include <seqan3/core/debug_stream.hpp>         // for debug_stream
+#include <seqan3/io/sequence_file/input.hpp>    // for sequence_file_input
+//! [fileinput_1]
+
+#include <filesystem>                           // to create temp directory
```
Comment (joergi-w) [inline doc/tutorial/introduction/introduction_file_input.cpp]:
```suggestion
#include <seqan3/std/filesystem>                // to create temp directory 
```

File: doc/tutorial/introduction/index.md (hunk 2 above)
Comment (joergi-w) [reply doc/tutorial/introduction/index.md]: Do you refer to the filesystem include? Please use instead:
```
#include <seqan3/std/filesystem>
```

**Pull request closed without merging**